### PR TITLE
KeepAliveCache Blocks Threads while Closing Connections

### PR DESCRIPTION
**Pull request by PoojaDP-23**
KeepAliveCache Blocks Threads while Closing Connections

### CODE REVIEW

**Comment by pshipton** [comment]
Backport of https://github.com/ibmruntimes/openj9-openjdk-jdk17/commit/77370e0b654c

File: src/java.base/share/classes/sun/net/www/http/KeepAliveCache.java [hunk 1]
```diff
@@ -157,19 +163,25 @@ public Void run() {
...
                     }
+                } else if (keepAliveTimeout == -2) {
+                    keepAliveTimeout = 0;
                 }
                 // at this point keepAliveTimeout is the number of seconds to keep
                 // alive, which could be 0, if the user specified 0 for the property
                 assert keepAliveTimeout >= 0;
                 if (keepAliveTimeout == 0) {
-                    http.closeServer();
+		     oldClient = http;
```
Comment (keithc-ca) [inline src/java.base/share/classes/sun/net/www/http/KeepAliveCache.java:173]: Please indent only with spaces.

File: src/java.base/share/classes/sun/net/www/http/KeepAliveCache.java [hunk 2]
```diff
@@ -280,36 +302,36 @@ class ClientVector extends ArrayDeque<KeepAliveEntry> {
...
+                         + Long.toString(currentTime - e.idleStartTime);
+                 KeepAliveCache.logger.finest(msg);
+             }
+             return e.hc;
+       }
     }
 
     /* return a still valid, unused HttpClient */
-    synchronized void put(HttpClient h) {
+    synchronized HttpClient put (HttpClient h) {
```
Comment (keithc-ca) [inline src/java.base/share/classes/sun/net/www/http/KeepAliveCache.java:325]: Please remove the space before `(`.

File: src/java.base/share/classes/sun/net/www/http/KeepAliveCache.java [hunk 3]
```diff
@@ -381,8 +403,8 @@ public int hashCode() {
 }
 
 class KeepAliveEntry {
-    HttpClient hc;
-    long idleStartTime;
+    final HttpClient hc;
+    final  long idleStartTime;
```
Comment (keithc-ca) [inline src/java.base/share/classes/sun/net/www/http/KeepAliveCache.java:407]: Please remove the extra space before `long`.

**Review by keithc-ca — CHANGES_REQUESTED** [review]
This includes changes not in the referenced commit from jdk17; in particular, the additions of lines 166-167.

**Comment by keithc-ca** [comment]
Including this change may be beneficial as well:
* [8229867: Re-examine synchronization usages in http and https protocol handlers](https://github.com/ibmruntimes/openj9-openjdk-jdk17/commit/f27a420dfd115f9a4721d4b70143ec13806c8643)

**Comment by narkedi** [comment]
@pshipton @keithc-ca This is actually not meant for merging to openj9 branch. Because we didn't had permissions to create the ifix branch, it got raised against the openj9 branch. Will request the developer to raise the PR against the ifix branch once its created. This is meant for delivering the ifixes requested against Semeru OE release.

**Comment by narkedi** [comment]
@pshipton @keithc-ca 
This PR can be closed as it not targeted for the intended branch. A new PR will be raised against the ifix branch which is now available. If there are no concerns, I will close this PR.

**Comment by keithc-ca** [comment]
It would be better to simply retarget this rather than separate the discussion from the eventual change.

**Pull request closed without merging**